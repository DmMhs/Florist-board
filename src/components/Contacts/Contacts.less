.Contacts {
  display: flex;
  flex-direction: column;
  align-items: center;
  .contacts-wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    h2 {
      font-size: 2rem;
      font-weight: normal;
      color: #333333;
    }
    hr {
      width: 75%;
      border: 0.5px solid #eee;
      margin: 1.5rem 0;
    }
  }
}
@media screen and (max-width: 768px) {
  .Contacts {
    margin-top: 4rem;
  }
}
