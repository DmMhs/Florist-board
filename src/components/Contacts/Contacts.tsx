import React from 'react';

const contacts = () => {
    return (
        <div className="Contacts">
        Contacts Component         
        </div>
    );
};

export default contacts;