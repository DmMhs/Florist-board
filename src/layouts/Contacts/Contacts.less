.Contacts {
  display: flex;
  flex-direction: column;
  align-items: center;
  min-height: calc(100vh - 7rem);
  .contacts-wrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    h2 {
      font-size: 2rem;
      font-weight: normal;
      color: #333333;
    }
    hr {
      width: 75%;
      border: 0.5px solid #eee;
      margin: 1.5rem 0;
    }
  }
}
@media screen and (max-width: 768px) {
  .Contacts {
    margin-top: 2.5rem;
  }
}
