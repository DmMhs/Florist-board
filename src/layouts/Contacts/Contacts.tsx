import React, { Component } from 'react';

import GoogleMap from './Map/GoogleMap';
import ContactInfo from './ContactInfo/ContactInfo';
import { AppContext } from '../../AppContext';
import { urls } from '../../config/urls';
import './Contacts.less';
import { getURLs } from '../../services/admin/getURLs';
import { getContacts } from '../../services/admin/getContacts';

interface ContactsProps {}
interface ContactsState {
  contacts: {
    en: {
      address: string;
    };
    ua: {
      address: string;
    };
    postCode: string;
    phone: string;
    email: string;
  };
  social_urls: {
    facebook: string;
    instagram: string;
    telegram: string;
  };
  google_map_url: string;
  fetchInProgress: boolean;
}

class Contacts extends Component<ContactsProps, ContactsState> {
  constructor(props: ContactsProps) {
    super(props);
    this.state = {
      contacts: {
        en: {
          address: ''
        },
        ua: {
          address: ''
        },
        postCode: '',
        phone: '',
        email: ''
      },
      social_urls: {
        facebook: '',
        instagram: '',
        telegram: ''
      },
      google_map_url: '',
      fetchInProgress: true
    };
  }

  public componentDidMount = async () => {
    const contacts = await getContacts();

    this.setState({
      contacts: contacts
    });
    const data = await getURLs();
    this.setState({
      google_map_url: data.google_map_address,
      social_urls: {
        facebook: data.socials.facebook,
        instagram: data.socials.instagram,
        telegram: data.socials.telegram
      },
      fetchInProgress: false
    });
  };

  public render() {
    const context = this.context;
    const labels = context.state.labels;
    const lang = context.state.lang;
    const contacts = this.state.contacts;
    const social = this.state.social_urls;
    return (
      <AppContext.Consumer>
        {value => (
          <div className="Contacts">
            <div className="contacts-wrapper">
              <h2>{labels[lang].pages.contacts.map}</h2>
              <GoogleMap url={urls.google_map_address} />
              <hr />
              <ContactInfo
                address={(contacts as { [key: string]: any })[lang].address}
                phone={contacts.phone}
                email={contacts.email}
                instagram={social.instagram}
                facebook={social.facebook}
                telegram={social.telegram}
              />
            </div>
          </div>
        )}
      </AppContext.Consumer>
    );
  }
}

Contacts.contextType = AppContext;

export default Contacts;
