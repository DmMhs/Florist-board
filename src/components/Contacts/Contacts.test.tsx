import React from 'react';
import { shallow } from 'enzyme';

import Contacts from './Contacts';

it('matches a snapshot', () => {
  const wrapper = shallow(<Contacts />);
  expect(wrapper).toMatchSnapshot();
});
