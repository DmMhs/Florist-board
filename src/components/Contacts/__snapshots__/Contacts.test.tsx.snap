// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts component works as expected matches a snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
